activation/poet.go

```diff
@@ -530,9 +543,31 @@ func (c *poetService) recertify(ctx context.Context, id types.NodeID) (*certifie
 	if c.certifier == nil {
 		return nil, errors.New("certifier not configured")
 	}
-	url, pubkey, err := c.client.CertifierInfo(ctx)
+	url, pubkey, err := c.getCertifierInfo(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("getting certifier info: %w", err)
+		return nil, err
 	}
 	return c.certifier.Recertify(ctx, id, url, pubkey)
 }
+
+func (c *poetService) getCertifierInfo(ctx context.Context) (*url.URL, []byte, error) {
+	c.certifierInfoMutex.Lock()
+	defer c.certifierInfoMutex.Unlock()
+	if time.Since(c.certifierInfo.obtained) <= c.certifierInfoTTL {
+		return c.certifierInfo.url, c.certifierInfo.pubkey, nil
+	}
+	url, pubkey, err := c.client.CertifierInfo(ctx)
+	if err != nil {
+		if !c.certifierInfo.obtained.IsZero() {
+			c.logger.Warn("failed to obtain new certifier info, falling back to the old value")
+			return c.certifierInfo.url, c.certifierInfo.pubkey, nil
+		}
```

Comment:
I don't think this is a good idea. If the TTL passed the value isn't valid any more. Imagine the szenario where a value is cached from 1 epoch ago and the request fails - the node would use info that might be outdated from 2 weeks ago. Wdyt?

Reply:
Generally, the certifier info shouldn't change at all. When it fails to obtain the info from the poet server, it's almost always safe to use the previous values. If the info turns out to be outdated, the following certification should fail and the retry mechanism will try to fetch the certifier info again (because `obtained` is not updated here and stays outdated).

Reply:
We know that non-official PoETs have been restarted in the past without preserving their keys. I think we shouldn't cache such info for an extended period of time to avoid issues that come from this. The truth comes from the `/info` endpoint - we shouldn't assume that the data returned from there doesn't change when calling the endpoint after an extended period of time.

Reply:
Makes sense :+1: thx